group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -411,9 +413,37 @@ public CompletableFuture<LeaveGroupResponseData> leaveGroup(
             return FutureUtils.failedFuture(Errors.COORDINATOR_NOT_AVAILABLE.exception());
         }
 
-        return FutureUtils.failedFuture(Errors.UNSUPPORTED_VERSION.exception(
-            "This API is not implemented yet."
-        ));
+        if (!isGroupIdNotEmpty(request.groupId())) {
+            return CompletableFuture.completedFuture(new LeaveGroupResponseData()
+                .setErrorCode(Errors.INVALID_GROUP_ID.code()));
+        }
+
+
+        return runtime.scheduleWriteOperation("generic-group-leave",
```

Comment:
nit: Let's put `"generic-group-leave"` on a new line in order to remain consistent with the existing code.

Reply:
all the old apis have the same issue. should i fix this in this PR?

Reply:
yes, you can.